nipoppy/extractors/networks/extract_network.py

```diff
@@ -0,0 +1,355 @@
+    DKT_DWI = Path(SEGPATH, f"sub-{participant_id}_ses-{session_id}_space-dwi_atlas-DKTatlas+aseg_dseg.nii.gz")
+
+    # paths to ANTs xfom files
+    ANAT2MNI_ALL = Path(FP_PATH, f"sub-{participant_id}", f"ses-{session_id}", 'anat', f"sub-{participant_id}_ses-{session_id}_run-1_from-T1w_to-MNI152NLin2009cAsym_mode-image_xfm.h5")
+    MNI2ANAT_ALL = Path(FP_PATH, f"sub-{participant_id}", f"ses-{session_id}", 'anat', f"sub-{participant_id}_ses-{session_id}_run-1_from-MNI152NLin2009cAsym_to-T1w_mode-image_xfm.h5")
+    FS2ANAT_AFF = Path(FP_PATH, f"sub-{participant_id}", f"ses-{session_id}", 'anat', f"sub-{participant_id}_ses-{session_id}_run-1_from-fsnative_to-T1w_mode-image_xfm.txt")
+
+    # path to space-dwi T1 image
+    ANAT_DWI = Path(TF_PATH, f"sub-{participant_id}", 'Register_T1', f"sub-{participant_id}__t1_warped.nii.gz")
+
+    # paths to tractoflow native alignment files
+    ANAT2DWI_AFF = Path(TF_PATH, f"sub-{participant_id}", 'Register_T1', f"sub-{participant_id}__output0GenericAffine.mat")
+    ANAT2DWI_SYN = Path(TF_PATH, f"sub-{participant_id}", 'Register_T1', f"sub-{participant_id}__output1Warp.nii.gz")
+    TRACTOGRAPHY = Path(TF_PATH, f"sub-{participant_id}", 'PFT_Tracking', f"sub-{participant_id}__pft_tracking_prob_wm_seed_0.trk")
+
+    # path to templateflow space / labels / reference volume
+    MNIPATH = Path(TEMPLATEFLOW_DIR, 'tpl-MNI152NLin2009cAsym')
+    MNITEMP = Path(MNIPATH, 'tpl-MNI152NLin2009cAsym_res-02_T1w.nii.gz')
+
+    # convert the DKT cortical parcellation from .mgz to .nii.gz
+    create_dkt_atlas(DKT_MGZ, DKT_NII)
+
+    # copy the full set of labels - Ideally prune down to just the cortical...
+    if ~os.path.exists(Path(LABPATH, f"atlas-DKT_dseg.tsv")):
+        print("Copy DKT labels file to output")
+        shutil.copyfile(Path(TEMPLATEFLOW_DIR, 'tpl-fsaverage', "tpl-fsaverage_dseg.tsv"),
+                        Path(LABPATH, f"atlas-DKT_dseg.tsv"))
+
+    # convert DKT cortical labels to MNI space
+    # subprocess.run(f'antsApplyTransforms -d 3 -e 0 -i {DKT_NII} -r {MNITEMP} -o {DKT_MNI} -n GenericLabel -v 1 -t {ANAT2MNI_ALL} {FS2ANAT_AFF}')
+    sub_dkt_mni = f'antsApplyTransforms -d 3 -e 0 -i {DKT_NII} -r {MNITEMP} -o {DKT_MNI} -n GenericLabel -v 1 -t {ANAT2MNI_ALL} {FS2ANAT_AFF}'
+    cmd_dkt_mni = sub_dkt_mni.split()
+    subprocess.run(cmd_dkt_mni)
+
+    # convert DKT cortical labels into participant DWI space
+    # subprocess.run(f'antsApplyTransforms -d 3 -e 0 -i {DKT_NII} -r {ANAT_DWI} -o {DKT_DWI} -n GenericLabel -v 1 -t {ANAT2DWI_SYN} {ANAT2DWI_AFF} {FS2ANAT_AFF}')
+    sub_dkt_dwi = f'antsApplyTransforms -d 3 -e 0 -i {DKT_NII} -r {ANAT_DWI} -o {DKT_DWI} -n GenericLabel -v 1 -t {ANAT2DWI_SYN} {ANAT2DWI_AFF} {FS2ANAT_AFF}'
+    cmd_dkt_dwi = sub_dkt_dwi.split()
+    subprocess.run(cmd_dkt_dwi)
+
+    # define Schaefer 2018 resolutions
+    schaeferRes = [100, 200, 300, 400, 500, 600, 800, 1000]
+    schaeferLab = [7, 17]
```

Comment:
It would be better if this can be set with an argument at runtime, with default options.